lib/ansible/modules/cloud/scaleway/scaleway_user_data.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python
+#
+# Scaleway user data management module
+#
+# Copyright (C) 2018 Online SAS.
+# https://www.scaleway.com
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+
+DOCUMENTATION = '''
+---
+module: scaleway_user_data
+short_description: Scaleway user_data management module
+version_added: "2.7"
+author: Remy Leone (@sieben)
+description:
+    - "This module manages user_data on compute instances on Scaleway."
+    - "It can be used to configure cloud-init for instance"
+options:
+
+  server_id:
+    description:
+    - Scaleway Compute instance ID of the server
+    required: true
+
+  oauth_token:
+    description:
+     - Scaleway OAuth token.
+    required: true
+
+  user_data:
+    description:
+    - User defined data. Typically used with `cloud-init`.
+    - Pass your cloud-init script here as a string
+    required: false
+
+  region:
+    description:
+    - Scaleway compute zone
+    required: true
+    choices:
+      - ams1
+      - EMEA-NL-EVS
+      - par1
+      - EMEA-FR-PAR1
+
+  timeout:
+    description:
+    - Timeout for API calls
+    required: false
+    default: 30
+'''
+
+EXAMPLES = '''
+- name: Update the cloud-init
+  scaleway_user_data:
+    server_id: '5a33b4ab-57dd-4eb6-8b0a-d95eb63492ce'
+    region: ams1
+    user_data:
+      cloud-init: 'final_message: "Hello World!"'
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import env_fallback
+from ansible.module_utils.scaleway import ScalewayAPI, SCALEWAY_LOCATION
+
+
+def patch_user_data(compute_api, server_id, key, value):
+    compute_api.module.debug("Starting patching user_data attributes")
+
+    path = "servers/%s/user_data/%s" % (server_id, key)
+    response = compute_api.patch(path=path, data=value, headers={"Content-Type": "text/plain"})
```

Comment:
To match the headers in https://github.com/sieben/ansible/blob/b119f2652a160707ee718566ad1e49a510935553/lib/ansible/module_utils/scaleway.py#L40
`Content-Type` must be `Content-type`